### ISSUE TITLE
tree: formatting AST nodes with quoted types does not work

### ISSUE THREAD

**Issue by otan** [cockroachdb/cockroach#93199]
Running `CREATE TABLE custom_type_quotes (a public."QuotedType")` through `TestParseDataDriven` would fail with 
```
        parse_test.go:77: /Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/parser/testdata/create_table:2464: mismatched AST when reparsing without literals:
            original: CREATE TABLE custom_type_quotes (a public."QuotedType")
            expected: CREATE TABLE custom_type_quotes (a public.QuotedType)
            actual:   CREATE TABLE custom_type_quotes (a public.quotedtype)
```

I've traced this down to `SQLString` using `FmtBareIdentifiers`, which produces an incorrect pretty print: https://github.com/cockroachdb/cockroach/blob/9804737e205f7de96c1f0383d90340b974e1ed18/pkg/sql/sem/tree/type_name.go#L257-L261

This was added in https://github.com/cockroachdb/cockroach/commit/dbdfcb74201addd90a907e111fdfb71b6464d875 where this remark was made:

> For example, the type DECIMAL would resolve to "DECIMAL", which cannot be parsed correctly to a valid type. 

But I fail to see why we need to support this - it does not work in PG:
```
otan=# create table t (a "DECIMAL");
ERROR:  type "DECIMAL" does not exist
LINE 1: create table t (a "DECIMAL");
```

Jira issue: CRDB-22217

**Comment by postamar** [comment]
@chengxiong-ruan this is not urgent, I thought perhaps this might interest you, put it back in the triage bucket if that's not the case.

**Comment by chengxiong-ruan** [comment]
This was fixed by #97514
